lib/KTable/index.vue

```diff
@@ -294,156 +294,146 @@
-                  return;
-                } else {
-                  if (colIndex > 0) {
-                    nextColIndex = colIndex - 1;
-                  } else if (rowIndex > 0) {
-                    nextColIndex = totalCols - 1;
-                    nextRowIndex = rowIndex - 1;
-                  } else {
-                    // Allow default behavior when reaching the first cell
-                    return;
-                  }
-                }
-              }
+        // Include the cell itself as the first focusable element
+        const cellAndFocusableElements = [currentCell, ...focusableElements];
+        const focusedIndex = cellAndFocusableElements.indexOf(focusedElement);
+
+        if (!event.shiftKey) {
+          // Tab key navigation
+          if (focusedIndex < cellAndFocusableElements.length - 1) {
+            // Move to the next focusable element within the cell
+            cellAndFocusableElements[focusedIndex + 1].focus();
+            event.preventDefault();
+            return;
+          } else {
+            // Move to the first focusable element of the next cell
+            if (colIndex < totalCols - 1) {
+              nextColIndex = colIndex + 1;
+            } else if (rowIndex < totalRows - 1) {
+              nextColIndex = 0;
+              nextRowIndex = rowIndex + 1;
             } else {
-              if (!event.shiftKey) {
-                if (colIndex < totalCols - 1) {
-                  nextColIndex = colIndex + 1;
-                } else if (rowIndex < totalRows - 1) {
-                  nextColIndex = 0;
-                  nextRowIndex = rowIndex + 1;
-                } else {
-                  // Allow default behavior when reaching the last cell
-                  return;
-                }
-              } else {
-                if (colIndex > 0) {
-                  nextColIndex = colIndex - 1;
-                } else if (rowIndex > 0) {
-                  nextColIndex = totalCols - 1;
-                  nextRowIndex = rowIndex - 1;
-                } else {
-                  // Allow default behavior when reaching the first cell
-                  return;
-                }
-              }
+              return; // Allow default Tab behavior when reaching the end
             }
-
-            break;
+            const nextCell = this.getCell(nextRowIndex, nextColIndex);
+            const nextFocusableElements = this.getFocusableElements(nextCell);
+            const nextCellAndFocusableElements = [nextCell, ...nextFocusableElements];
+            nextCellAndFocusableElements[0].focus();
+            this.updateFocusState(nextRowIndex, nextColIndex);
+            event.preventDefault();
           }
-
-          default:
+        } else {
+          // Shift+Tab key navigation
+          if (focusedIndex > 0) {
+            // Move to the previous focusable element within the cell
+            cellAndFocusableElements[focusedIndex - 1].focus();
+            event.preventDefault();
```

Comment:
This condition would need to be changed to -

```
 if (focusedIndex < cellAndFocusableElements.length - 1) {
            // Move to the previous focusable element within the cell
            cellAndFocusableElements[focusedIndex + 1].focus();
            event.preventDefault();
            return;
          } 

```
(Same as when navigating using tab key, since there is some ambiguity caused with the focusCell function when the current condition is used)
You can have a look at it yourself - like if you don't change the shift+tab key condition here with the tab key condition and add `this.focusCell` to `updateFocusState`, the focusable elements within cells would no longer be focused when we navigate using Shift+Tab keys.

Reply:
@BabyElias Yes, I did it myself and it was having problem with `this.focusCell` as you said. Now it's fixed and working fine as expected.

I have a bit of confusion regarding the navigation order when using `shift+tab` to move back to a previous cell and the focusable elements within that cell.

For instance, given the following layout:

- row n: cell1, cell2, cell3 > (button1, button2)
- row n+1: cell1, cell2, cell3 > (button1, button2)

If the focus is on `cell1` of row[n+1] and `shift+tab` is pressed, should the focus move to:
1. `cell3` of row[n]?
2. `button1` or `button2` within `cell3` of row[n]?
Or should it follow the sequence: `cell3` → `button1` → `button2` and so on? (this is our current implementation)

Another example:
- row1: cell1, cell2, cell3 > (button1, button2)
- row2: cell1, cell2, cell3 > (button1, button2)

Suppose the focus is on `cell3` of row1 (which contains focusable elements) and we press `shift+tab`. Should the focus move to `cell2` or `button1`/`button2` ? Personally, I suggest that in this case, the focus should move to `cell2` instead of `button1`, as moving directly to the button could lead to a slightly awkward user experience when navigating backward using `shift+tab`.

For now, I’ve implemented it as you suggested. Please let me know if there’s anything wrong with it. :)

Reply:
For the first layout, the cell should follow `cell3 → button1 → button2 `, as in the current implementation - as a user should only be able to move to the focusable elements when tab key is pressed, otherwise tab and arrow keys should work the same way.
For the second layout, ideally it should be `cell2→button1/button2` when using Shift+Tab key - which is not how it currently works in our implementation. This is something which I had tried figuring out earlier as well but eventually it just lost direction. If you are able to implement this expected nav behaviour with `focusCell` functionality intact - that would be just perfect. No pressure around it, in case you wish to leave it as it is. 
Thank you for your time and efforts!

Reply:
> ideally it should be `cell2→button1/button2` when using Shift+Tab key - which is not how it currently works in our implementation.

I'm in bit confusion. Do you mean if we are `cell3` of row1 then pressing `shift+tab` should be move to `cell2` or `button1/button2`? Because what I'm infering from what you wrote that line is that if our current focus is at `cell3` then pressing `shift+tab` should move to `cell2` then `button1/button2` of `cell3`.

Reply:
>as the purpose of Shift+Tab key is to move in the exact reverse order as Tab key (opposite in DOM order) and so the movement from current cell to prev cell button to prev cell is most ideal. 

Thanks for getting it. That is exactly 💯  what I was trying to say. 
So, should I go for it now (which may take couple of days or more)? or Should we leave it as it for now and open a new issue which can be solved later anytime?

Reply:
In case you are willing to do it yourself, doing it in this PR itself irrespective of the time works! Otherwise we can also open up a follow-up issue too . Anything that suits you.

Reply:
I am fine leaving you to decide, just a recommendation - usually whenever follow-up issue is possible, it's generally good practice. Quite often some unforeseen things come up that potentially block work that otherwise could be merged, and also reviews and QA are easier if it can be done in smaller chunks.

One way or another, we're happy to have you do that update @shivam-daksh :)